test/unit/locator_test.js

```diff
@@ -85,6 +85,13 @@ describe('Locator', () => {
         expect(l.toString()).to.equal('foo');
       });
 
+      it('should create shadow locator', () => {
+        const l = new Locator({ shadow: '["my-app", "recipe-hello-binding", "ui-input", "input.input"]' });
```

Comment:
sorry, I still don't understand why do you pass it as string if you can pass it by array:

```js
{ shadow: ["my-app", "recipe-hello-binding", "ui-input", "input.input"] }
```